File: codewars/7kyu/selective-fear-of-numbers/dduck-selective-fear-of-numbers.py
// https://www.codewars.com/kata/55b1fd84a24ad00b32000075/

from unittest import TestCase


def am_I_afraid(day, num):
    days = {
        "Monday": lambda n: n == 12,
        "Tuesday": lambda n: n > 95,
        "Wednesday": lambda n: n == 34,
        "Thursday": lambda n: n == 0,
        "Friday": lambda n: n % 2 == 0,
        "Saturday": lambda n: n == 56,
        "Sunday": lambda n: n == 666 or n == -666,
    }
    return days[day](num)


TestCase().assertEqual(am_I_afraid("Monday", 13), False, 'Should return false')
TestCase().assertEqual(am_I_afraid("Sunday", -666), True, 'Should return true')
TestCase().assertEqual(am_I_afraid("Tuesday", 2), False, 'Should return false')
TestCase().assertEqual(am_I_afraid("Tuesday", 965), True, 'Should return true')
TestCase().assertEqual(am_I_afraid("Friday", 2), True, 'Should return true')

File: codewars/7kyu/selective-fear-of-numbers/dduck-selective-fear-of-numbers.js
// https://www.codewars.com/kata/55b1fd84a24ad00b32000075/

const AmIAfraid = function (day, num) {
  switch (day) {
    case 'Monday':
      return num === 12;
    case 'Tuesday':
      return num > 95;
    case 'Wednesday':
      return num === 34;
    case 'Thursday':
      return num === 0;
    case 'Friday':
      return num % 2 === 0;
    case 'Saturday':
      return num === 56;
    default:
      return num === 666 || num === -666;
  }
};

console.log(AmIAfraid('Monday', 13), false, 'Should return false');
console.log(AmIAfraid('Sunday', -666), true, 'Should return true');
console.log(AmIAfraid('Tuesday', 2), false, 'Should return false');
console.log(AmIAfraid('Tuesday', 965), true, 'Should return true');
console.log(AmIAfraid('Friday', 2), true, 'Should return true');
